### PR TITLE
Add InversionNet speed-of-sound reconstruction model

### PR DESCRIPTION
**Pull request by tristan-deep**
Adds `zea.models.inversionnet.InversionNet`: a Keras 3 port of the pretrained [InversionNet](https://doi.org/10.1109/TCI.2019.2956866) baseline released with the [OpenPros](https://open-pros.github.io/) limited-view prostate USCT benchmark. It reconstructs a speed-of-sound map directly from multi-source waveform data.

The original is PyTorch-only. The port runs on JAX, TensorFlow and PyTorch, and reproduces the original output to within float32 round-off (max abs. difference 3.6e-5 on the real weights, verified on CPU).

```python
model = InversionNet.from_preset("inversionnet-openpros")
sos = model(waveforms)  # (B, 1000, 161, 40) -> (B, 401, 161, 1), in [-1, 1]
```

**What's here**

- `zea/models/inversionnet.py` — the model, plus `load_torch_state_dict()` / `from_pth()` for converting the original checkpoint.
- Preset `inversionnet-openpros` → [`zeahub/openpros-inversion-net`](https://huggingface.co/zeahub/openpros-inversion-net) (weights, config, and the original `.pth` for provenance).
- Tests on a miniature configuration, since the real one is 20M parameters. The default architecture is checked symbolically against the released checkpoint's output shape and parameter count.
- `docs/source/models.rst` — a few corrections to the PyTorch conversion guide from doing this port: `BatchNorm` epsilon and weight order, the general `ConvTranspose` padding rule, why layer classes must not start with an underscore, and a warning that TF32 on GPU makes a correct conversion look broken.

This lets the OpenPros [OpenH-RF submission](https://github.com/open-h/OpenH-RF) drop its vendored network and its 82 MB checkpoint and just load the preset.

## Summary by CodeRabbit

* **New Features**
  * Added the InversionNet model for reconstructing ultrasound computed tomography speed-of-sound maps.
  * Added a pretrained OpenPros InversionNet preset with support for loading model weights.
  * Made InversionNet available through the models package.

* **Documentation**
  * Expanded custom model porting guidance, including padding, normalization, symbolic construction, validation, and ONNX workflows.
  * Added InversionNet as a documented worked example.

* **Tests**
  * Added coverage for model outputs, input validation, configuration serialization, architecture compatibility, batch normalization behavior, and preset weight loading.

### CODE REVIEW

**Comment by swpenninga** [comment]
I found the port is identical to the original for `jax`, `tensorflow` and `torch`.

The model currently does not work with `model.fit()` though (BatchNorm is frozen), but I assume retraining is never going to happen, since we cite their model.

A few small things claude found:
- Longer inputs (2000,161,40) return (1425,161,1) and raise no error or warning. 
Proposed fix:
```python
for block in self.encoder:
    x = block(x)
if tuple(x.shape[1:3]) != (1, 1):
    raise ValueError(
        f"The encoder must collapse to 1 x 1 before the decoder, but produced "
        f"{tuple(x.shape[1:3])}. bottle_conv={self.bottle_conv} does not match the "
        f"feature map that waveform_shape={self.waveform_shape} yields; the decoder "
        f"would upsample the leftover extent and silently return a wrongly sized map."
    )
```
- Input preprocessing is unreproducible from the docs. I don't quite understand the model specifics, but it seems that `k` and `data_min/data_max` need to be chosen and a sweep showed very different results:
image
image

**Comment by tristan-deep** [comment]
Thanks @swpenninga ! I implemented those suggestions, also made `model.fit()` work. Even though it is mainly and inference model it would be a pity if it could not be retrained / repurposed at all. Ready for review!

Then regarding the input processing, I clarified this. But this is also an ongoing issue, see here [our roadmap](https://github.com/tue-bmd/zea/issues/48) under models.

**Review by swpenninga — APPROVED** [review]
